javascripts/discourse/components/custom-category-toggle.gjs

```diff
@@ -0,0 +1,77 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import { NotificationLevels } from "discourse/lib/notification-levels";
+import I18n from "discourse-i18n";
+
+export default class CustomCategoryToggle extends Component {
+  @service currentUser;
+
+  @tracked isLoading = false;
+
+  category = this.args.outletArgs.category;
```

Comment:
unless there is a specific need for a "frozen in time" arg, it's discouraged to assign an arg to a prop like this (because this.category wouldn't get updated when the arg changes)

```suggestion
  get category() {
    return this.args.outletArgs.category;
  }
```